python/packages/autogen-agentchat/src/autogen_agentchat/agents/_base_chat_agent.py

```diff
@@ -39,10 +39,15 @@ class BaseChatAgent(ChatAgent, ABC, ComponentBase[BaseModel]):
     component_type = "agent"
 
     def __init__(self, name: str, description: str) -> None:
-        self._name = name
-        if self._name.isidentifier() is False:
-            raise ValueError("The agent name must be a valid Python identifier.")
-        self._description = description
+        """Initialize the agent with a name and description."""
+        with trace_create_agent_span(
```

Comment:
I notice this doesn't pass the `agent_id`, is that because it doesn't exist yet?

Reply:
Yes. Though I could try to grab it using AgentInstantiationContext. This addresses the case when the AgentChat agent is created inside a Core Agent's constructor. However, when agent is created outside of the CoreAgent there isn't an id, because agent id is associated with the underlying runtime.